core/src/main/java/org/infinispan/statetransfer/StateConsumerImpl.java

```diff
@@ -502,6 +507,32 @@ public boolean ownsData() {
       });
    }
 
+   private IntSet computeTransactionOnlySegments(CacheTopology cacheTopology, ConsistentHash newReadCh, Address address) {
+      IntSet transactionOnlySegments = IntSets.mutableEmptySet(numSegments);
+      if (configuration.transaction().lockingMode() == LockingMode.PESSIMISTIC &&
```

Comment:
checking for `lockingMode` is not enough IMO. User can set the `LockingMode` without enabling the `TransactionMode`

Reply:
I would also revert the if-condition to avoid having the code idented 
```java
      if (configuration.transaction().transactionMode() == TransactionMode.NON_TRANSACTIONAL ||
            configuration.transaction().lockingMode() == LockingMode.OPTIMISTIC ||
            cacheTopology.getPendingCH() == null ||
            !cacheTopology.getCurrentCH().getMembers().contains(address)) {
         return IntSets.immutableEmptySet();
      }
      ... continue with computation 
```

Reply:
Good catch